pages/api/user/index.ts

```diff
@@ -18,10 +21,29 @@ export default async function handle(
             break;
         }
         case "POST": {
+            // TODO
             break;
         }
         case "PUT": {
-            // TODO:
+            const session = await getSession({ req });
+            const userId = session.id as string;
+            // TODO: add user role to session
+            const updatedUserData = {
+                id: userId,
+                firstName: req.body.firstName,
+                lastName: req.body.lastName,
+                role: req.body.role,
+                roleData: req.body.roleData,
+            } as UserInput;
+            const updatedUser = await updateUser(updatedUserData);
```

Comment:
are we not doing any validation before updating the user data?

Reply:
This is happening in a separate PR

Reply:
See here: https://github.com/uwblueprint/social-diversity-for-children/pull/35